Backend Code/VigyBag/users/serializers.py

```diff
@@ -0,0 +1,43 @@
+from rest_framework import serializers
+from .models import CustomUser
+from django.contrib.auth import authenticate
+
+class RegisterUserSerializer(serializers.ModelSerializer):
+    email=serializers.EmailField(max_length=255)
+    password=serializers.CharField(max_length=128,write_only=True)
+
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')  
+        #have to add usertype ,default customer
+    
+    def create(self, validated_data):
+        user = CustomUser.objects.create_user(email=validated_data['email'], password=validated_data['password'])
+        return user
+    
+class LoginSerializer(serializers.ModelSerializer):
+    email=serializers.EmailField(max_length=255)
+    password=serializers.CharField(max_length=128,write_only=True)
+    class Meta:
+        model = CustomUser
+        fields = ('email', 'password')
+
+    def validate(self, data):
+        email=data.get('email', None)
+        password=data.get('password', None)
+        if email and password:
+            user=authenticate(email=email, password=password)
```

Comment:
This is again a call to function of the buisness logic layer that serializer should not be making.
Suggesting to move business logic and triggers to business logic out of serializers.